src/app/WirelessIno/app/src/main/java/se/sics/sse/fresta/wirelessino/Main.java

```diff
@@ -5,117 +5,280 @@
-		if (item.getItemId() == EXIT_INDEX) {
-			finish();
-		} 
-		else if (item.getItemId() == DISCONNECT_INDEX) {
-			try {
-				if (socket != null) {
-					socket.close();
-					socket = null;
-					
-					menu.getItem(DISCONNECT_INDEX).setVisible(false); // Hide the disconnect option
-					view.invalidate(); // Repaint (to show "not connected" in the main view)
-				}
-			} catch (IOException e) {
-				e.printStackTrace();
-			}
-		} 
-		else if (item.getItemId() == CONFIG_INDEX) {
-			Intent i = new Intent(Main.this, SocketConnector.class);
-			startActivity(i);
-		}
-		
-		return super.onOptionsItemSelected(item);
-	}
-
-	/* 
-	 * Initialize the output stream for the socket.  
-	 */
-	public static void init(Socket socket) {
-		Main.socket = socket;
-		try {
-			out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(
-					socket.getOutputStream())), true);
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
-	}
-
-	/*
-	 * Send a message through the socket.
-	 */
-	public static void send(Object message) {
-		out.println(message);
-	}
-	
-	/*
-	 * Checks if a socket connection has been established and updates
-	 * the visibility of the "disconnect" menu option accordingly.    
-	 */
-	private void updateMenuVisibility() {
-		if (menu != null) {
-			if (socket == null || !socket.isConnected()) 
-				menu.getItem(DISCONNECT_INDEX).setVisible(false);
-			else
-				menu.getItem(DISCONNECT_INDEX).setVisible(true);
-		}
-	}
+        updateMenuVisibility();
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    /**
+     * Handle different menu options
+     *
+     * @param item the pressed menu item
+     */
+
+
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == EXIT_INDEX) {
+            finish();
+        } else if (item.getItemId() == DISCONNECT_INDEX) {
+            try {
+                if (socket != null) {
+                    socket.close();
+                    socket = null;
+
+                    menu.getItem(DISCONNECT_INDEX).setVisible(false); // Hide the disconnect option
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else if (item.getItemId() == CONFIG_INDEX) {
+            Intent i = new Intent(Main.this, SocketConnector.class);
+            startActivity(i);
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    /**
+     * Initialize the output stream for the socket.
+     */
+    public static void init(Socket socket) {
+        Main.socket = socket;
+        try {
+            out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(
+                    socket.getOutputStream())), true);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Send a message through the socket.
+     *
+     * @param message the message that will be sent to the MOPED
+     */
+    public static void send(Object message) {

```

Comment:
Could have a more describing name like sendSocket or something.

Reply:
Dock skickas väl bara ett meddelande via socketen? Men kanske sendToMoped?

Reply:
True! I'm ok with that